src/vs/base/browser/ui/grid/grid.ts

```diff
@@ -455,7 +455,8 @@ export class Grid<T extends IView = IView> extends Disposable {
 		if (sizing?.type === 'distribute') {
 			gridViewSizing = GridViewSizing.Distribute;
 		} else if (sizing?.type === 'auto') {
-			gridViewSizing = GridViewSizing.Auto(0);
+			const index = location[location.length - 1];
+			gridViewSizing = GridViewSizing.Auto(index === 0 ? 1 : index - 1);
 		}
 
 		this.gridview.removeView(location, gridViewSizing);
```

Comment:
To be honest, I have no clue what this index transformation is for :D
Maybe a comment here could be useful :)

Reply:
Comments are on the API:

https://github.com/microsoft/vscode/blob/cdf84c668586456228a270489ac744ad61c3b09a/src/vs/base/browser/ui/splitview/splitview.ts#L340-L345